### ISSUE TITLE
Externalize OntoPortal client functions (e.g., for BioPortal, AgroPortal, EcoPortal) 

### ISSUE THREAD

**Issue by cthoyt** [INCATools/ontology-access-kit#168]
I have previous written a harness for accessing the API for OntoPortal instances in the Bioregistry project. I spun that code out into a clean project at https://github.com/cthoyt/ontoportal-client. I'd be happy to move that to a different namespace if other people want to contribute to it.

**Comment by pkalita-lbl** [comment]
Hey @cthoyt I took a look at `ontoportal-client` and it seems like it could drop in pretty easily. I did notice that there was a [recent change](https://github.com/cthoyt/ontoportal-client/blob/3bb6162656bddb24dcf072f9f07c3ffdbbdf99f2/src/ontoportal_client/api.py#L104) to make the client name a class variable. It turns out I think that would actually help make the integration into OAK a bit easier. Wondering if you have plans to make a release with those changes any time in the near future?

**Comment by cthoyt** [comment]
Yes I can drop a release anytime, it’s quite easy with the Tox configuration

**Comment by pkalita-lbl** [comment]
Here's a draft PR: https://github.com/INCATools/ontology-access-kit/pull/204 (still have some tests to fix!) 

Having the `name` class variable would just simply this bit [here](https://github.com/INCATools/ontology-access-kit/pull/204/files#diff-ef05ebc2897885968012ec9b30e4f5cbfda8d3e526d985ba2c469fce96081b3bR53). `OntoPortalInterface` could use that name for the API key lookup instead of having implementation classes [define it](https://github.com/INCATools/ontology-access-kit/pull/204/files#diff-48d790a8f0e29cc1eb533e007849cfc9467d8cdaecbded167e633b818caf121eR16) in addition to the `OntoPortalClient` class to use.

It's a fairly minor difference so if you want to make a release I'll gladly pick up the new version, otherwise I don't see much harm in updating it later.

**Comment by cmungall** [comment]
thanks!

reopening as we still only using the inner layer of ontoportal-client (the bit that returns json). 

there is still more to be done to move some parts of oaklib.impl.ontoportal into ontoportal-client, see:

https://github.com/cthoyt/ontoportal-client/issues/3

but if you like you can reclose this and open more specific issues, e.g for different components like annotator/search/ancestors
